deepspeed/runtime/zero/stage3.py

```diff
@@ -1686,10 +1690,14 @@ def get_grad_norm_direct(self, gradients, params, norm_type=2):
 
             self._model_parallel_all_reduce(tensor=total_norm_cuda, op=dist.ReduceOp.SUM)
 
-            total_norm = total_norm_cuda.item()**(1. / norm_type)
+            total_norm = total_norm_cuda**(1. / norm_type)
+
+        norm_is_inf = total_norm.isinf()
+        norm_is_nan = total_norm.isnan()
+        inf_or_nan = norm_is_nan.logical_or(norm_is_inf)
 
-        if total_norm == float('inf') or total_norm == -float('inf') or total_norm != total_norm:
-            total_norm = -1
+        err = torch.tensor(-1.0, device=self.current_device, dtype=torch.float)
```

Comment:
I don't think there is a `self.current_device` defined

Reply:
```suggestion
        err = torch.tensor(-1.0, device=self.device, dtype=torch.float)
```

Great catch, Mike!